In [1]:
import os
print(os.getcwd())

C:\Users\장세환\Desktop\회귀분석\빅콘테스트


In [2]:
os.chdir("E:/bigcontest")
print(os.getcwd())

E:\bigcontest


In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import csv

In [4]:
import datetime as dt
import seaborn as sns

In [5]:
data = pd.read_excel('./2020빅콘테스트 문제데이터(데이터분석분야-챔피언리그)/01_제공데이터/2020_bigcontest_champion_performance.xlsx',header=1)


In [5]:
data.head()

방송일시  노출(분)    마더코드    상품코드            상품명 상품군   판매단가  \
0 2019-01-01 06:00:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
1 2019-01-01 06:00:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
2 2019-01-01 06:20:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
3 2019-01-01 06:20:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
4 2019-01-01 06:40:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   

         취급액  
0  2099000.0  
1  4371000.0  
2  3262000.0  
3  6955000.0  
4  6672000.0

In [8]:
data['DATE'] = data['방송일시'].map(lambda x: 10000*x.to_pydatetime().year + 100*x.to_pydatetime().month + x.to_pydatetime().day)

In [7]:
data.head()

방송일시  노출(분)    마더코드    상품코드            상품명 상품군   판매단가  \
0 2019-01-01 06:00:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
1 2019-01-01 06:00:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
2 2019-01-01 06:20:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
3 2019-01-01 06:20:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
4 2019-01-01 06:40:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   

         취급액      DATE  
0  2099000.0  20190101  
1  4371000.0  20190101  
2  3262000.0  20190101  
3  6955000.0  20190101  
4  6672000.0  20190101

In [9]:
set(data['상품군'])

{'가구', '가전', '건강기능', '농수축', '생활용품', '속옷', '의류', '이미용', '잡화', '주방', '침구'}

In [10]:
grouped_day_item = data.groupby(['상품군','DATE']).sum()
grouped_day_item

노출(분)     마더코드     상품코드      판매단가        취급액
상품군 DATE                                                  
가구  20190201   60.0   901845  1806588   5382000  169247000
    20190204   60.0   901845  1806588   5382000  150677000
    20190206   60.0   901845  1806588   5382000  178832000
    20190208   60.0   901845  1806588   5382000  217876000
    20190212   60.0  1809000  3627108  10332000  164555000
...             ...      ...      ...       ...        ...
침구  20191111   60.0  1201716  2404746    898800   96350000
    20191113   60.0  1201716  2404746    898800  108818000
    20191119   60.0  1201716  2404746    898800  110848000
    20191121   60.0  1201716  2404746    898800  122006000
    20191124   60.0  1201716  2404746    898800  108289000

[2984 rows x 5 columns]

In [11]:
grouped_day_item = grouped_day_item['취급액']/grouped_day_item['판매단가']

In [12]:
#grouped_day_item = pd.DataFrame(grouped_day_item)
#grouped_day_item

In [12]:
grouped_day_item

상품군  DATE    
가구   20190201     31.446860
     20190204     27.996470
     20190206     33.227796
     20190208     40.482349
     20190212     15.926732
                    ...    
침구   20191111    107.198487
     20191113    121.070316
     20191119    123.328883
     20191121    135.743213
     20191124    120.481753
Length: 2984, dtype: float64

In [7]:

data['DATE'] = data['방송일시'].map(lambda x: 10000*x.to_pydatetime().year + 100*x.to_pydatetime().month + x.to_pydatetime().day)


In [8]:

data.head()


방송일시  노출(분)    마더코드    상품코드            상품명 상품군   판매단가  \
0 2019-01-01 06:00:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
1 2019-01-01 06:00:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
2 2019-01-01 06:20:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
3 2019-01-01 06:20:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
4 2019-01-01 06:40:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   

       취급액      DATE  
0  2099000  20190101  
1  4371000  20190101  
2  3262000  20190101  
3  6955000  20190101  
4  6672000  20190101

In [17]:
compare_2019=pd.concat([data["DATE"],data["상품군"]],axis=1)

In [21]:
compare_2019.head()

DATE 상품군
0  20190101  의류
1  20190101  의류
2  20190101  의류
3  20190101  의류
4  20190101  의류

In [19]:
len(compare_2019)

37372

In [38]:
panda_2019 = pd.DataFrame(compare_2019)

In [36]:
panda_2019.head()

DATE 상품군
0  20190101  의류
1  20190101  의류
2  20190101  의류
3  20190101  의류
4  20190101  의류

In [39]:
panda_2019['DATE'][0]

20190101

In [52]:
data_2020=pd.read_excel('compare_testset_dist.xlsx')

In [53]:
data_2020

카테고리        비율
0    의류   0.055690
1    속옷   0.113801
2   이미용   0.057765
3     주방  0.094085
4    농수축  0.146662
5    무형   0.169146
6     가전  0.169146
7   생활용품  0.120374
8     가구  0.054998
9     잡화  0.095815
10  건강기능  0.006226
11    침구  0.024905

In [133]:
data_2020['compare']=0
data_2020

카테고리        비율  compare
0    의류   0.055690        0
1    속옷   0.113801        0
2   이미용   0.057765        0
3     주방  0.094085        0
4    농수축  0.146662        0
5    무형   0.169146        0
6     가전  0.169146        0
7   생활용품  0.120374        0
8     가구  0.054998        0
9     잡화  0.095815        0
10  건강기능  0.006226        0
11    침구  0.024905        0

In [134]:
panda_2019['DATE'][0]

20190101

In [170]:
a=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i]) == '20190201':
        break
    else:
        a+=1
        if panda_2019['상품군'][i] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i] == '침구':
            data_2020['compare'][11]+=1     
            

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\

In [171]:
a

2877

In [172]:
data_2020['compare']

0     372
1     389
2      93
3     742
4     458
5       0
6     570
7      83
8       0
9     107
10     63
11      0
Name: compare, dtype: int64

In [173]:
month1=data_2020['비율']-data_2020['compare']/a

In [139]:
month1

0    -0.073611
1    -0.021409
2     0.025440
3    -0.163822
4    -0.012532
5     0.169146
6    -0.028977
7     0.091524
8     0.054998
9     0.058623
10   -0.015672
11    0.024905
dtype: float64

In [174]:
month_1=0
for i in range(len(month1)):
    if data_2020['compare'][i] != 0:
        month_1+=abs(month1[i])
month_1

0.491610426182103

In [89]:
len(panda_2019)

37372

In [100]:
a

2877

In [141]:
b=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a]) == '20190301':
        break
    else:
        b+=1
        if panda_2019['상품군'][i+a] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a] == '침구':
            data_2020['compare'][11]+=1     

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:1

In [142]:
month2=data_2020['비율']-data_2020['compare']/b
month_2=0
for i in range(len(month2)):
    month_2+=abs(month2[i])
month_2

0.6528915325311161

In [149]:
c=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b]) == '20190401':
        break
    else:
        c+=1
        if panda_2019['상품군'][i+a+b] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b] == '침구':
            data_2020['compare'][11]+=1     

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on

In [150]:
month3=data_2020['비율']-data_2020['compare']/c
month_3=0
for i in range(len(month3)):
    month_3+=abs(month3[i])
month_3

0.6619756318658376

In [151]:
d=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c]) == '20190501':
        break
    else:
        d+=1
        if panda_2019['상품군'][i+a+b+c] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c] == '침구':
            data_2020['compare'][11]+=1     

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is t

In [152]:
month4=data_2020['비율']-data_2020['compare']/d
month_4=0
for i in range(len(month4)):
    month_4+=abs(month4[i])
month_4

0.735957891079774

In [153]:
e=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d]) == '20190601':
        break
    else:
        e+=1
        if panda_2019['상품군'][i+a+b+c+d] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d] == '침구':
            data_2020['compare'][11]+=1     

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCop

In [154]:
month5=data_2020['비율']-data_2020['compare']/e
month_5=0
for i in range(len(month5)):
    month_5+=abs(month5[i])
month_5

0.6762211430402479

In [155]:
f=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e]) == '20190701':
        break
    else:
        f+=1
        if panda_2019['상품군'][i+a+b+c+d+e] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '침구':
            data_2020['compare'][11]+=1     
month6=data_2020['비율']-data_2020['compare']/f
month_6=0
for i in range(len(month6)):
    month_6+=abs(month6[i])
month_6            

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCop

0.6507850093299086

In [164]:
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f]) == '20190801':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '침구':
            data_2020['compare'][11]+=1     
month7=data_2020['비율']-data_2020['compare']/g
month_7=0
for i in range(len(month7)):
    month_7+=abs(month7[i])
month_7            

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: Setting

0.5388714731921482

In [165]:
h=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+g]) == '20190901':
        break
    else:
        h+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+g] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g] == '침구':
            data_2020['compare'][11]+=1     
month8=data_2020['비율']-data_2020['compare']/h
month_8=0
for i in range(len(month8)):
    month_8+=abs(month8[i])
month_8 

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:9: SettingWithCopyWarning: 
A

0.4544496142767479

In [166]:
i=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+g+h]) == '20191001':
        break
    else:
        i+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h] == '침구':
            data_2020['compare'][11]+=1     
month9=data_2020['비율']-data_2020['compare']/i
month_9=0
for i in range(len(month9)):
    month_9+=abs(month9[i])
month_9

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a Da

0.5957199637130168

In [167]:
j=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+g+h+i]) == '20191101':
        break
    else:
        j+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i] == '침구':
            data_2020['compare'][11]+=1     
month10=data_2020['비율']-data_2020['compare']/j
month_10=0
for i in range(len(month10)):
    month_10+=abs(month10[i])
month_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a Da

0.5365839184891514

In [168]:
k=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+g+h+i+j]) == '20191201':
        break
    else:
        k+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j] == '침구':
            data_2020['compare'][11]+=1     
month11=data_2020['비율']-data_2020['compare']/k
month_11=0
for i in range(len(month11)):
    month_11+=abs(month11[i])
month_11

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:21: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:29: SettingWithCopyWarning: 


0.4889311656866125

In [169]:
l=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+g+h+i+j+k]) == '20200101':
        break
    else:
        l+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+g+h+i+j+k] == '침구':
            data_2020['compare'][11]+=1     
month12=data_2020['비율']-data_2020['compare']/l
month_12=0
for i in range(len(month12)):
    month_12+=abs(month12[i])
month_12

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Anaconda\lib\site-packages\ipykernel_

0.5119695551002179

# 상품군별 test set과의 차이
month_1
0.491610426182103

month_2
0.6528915325311161

month_3
0.6619756318658376

month_4
0.735957891079774

month_5
0.6762211430402479

month_6
0.6507850093299086

month_7
0.5388714731921482

month_8
0.4544496142767479

month_9
0.5957199637130168

month_10
0.5365839184891514

month_11
0.4889311656866125

month_12
0.5119695551002179

#테스트 셋 마더코드 중복비율(월별로)                  
1 _Mothercode   
0.20839469808541974
Dataframe 2019_2_ratio created.

2 _Mothercode   
0.21391752577319587
Dataframe 2019_3_ratio created.

3 _Mothercode   
0.28829160530191456
Dataframe 2019_4_ratio created.

4 _Mothercode   
0.29491899852724596
Dataframe 2019_5_ratio created.

5 _Mothercode   
0.3229013254786451
Dataframe 2019_6_ratio created.

6 _Mothercode   
0.3015463917525773
Dataframe 2019_7_ratio created.

7 _Mothercode   
0.38181148748159055
Dataframe 2019_8_ratio created.

8 _Mothercode   
0.36892488954344627
Dataframe 2019_9_ratio created.

9 _Mothercode   
0.33505154639175255
Dataframe 2019_10_ratio created.

10 _Mothercode   
0.3317378497790869
Dataframe 2019_11_ratio created.

11 _Mothercode   
0.35162002945508103
Dataframe 2019_12_ratio created.

12 _Mothercode   
0.3634020618556701  

# 그리드 서치
7,8월 +10,

10+7,8월,

7,8월 +20,

20+7,8월,

7,8월 -10,

-10+7,8월 ,

7,8월 -20,

-20+7,8월,

10+7,8월+10,

20+7.8월+20

In [176]:
# 7,8월 +10

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f]) == '20190911':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10





C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:19: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:29: SettingWithCopyWarning: 

0.4609374371881794

In [191]:
# 10+7,8월 

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-10]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-10] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10


C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:19: SettingWithCopyWarning: 

0.4504118950753935

In [181]:
# 7,8월 +20

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f]) == '20190921':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:19: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:29: SettingWithCopyWarning: 

0.4832054779883091

In [192]:
#+20+7,8월 

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-20]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-20] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of

0.4516342309489186

In [184]:
# 7,8월 -10,
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f]) == '20190822':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:28: SettingWithCopyWarning: 
A value is trying to be set on 

0.5004110551541192

In [193]:
# -10+ 7,8월 
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+10]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+10] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+10] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:28: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:24: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a c

0.45228222021455866

In [186]:
# 7,8월 -20,

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f]) == '20190821':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:19: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:29: SettingWithCopyWarning: 

0.5083751540965035

In [194]:
# -20+7,8월,

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f+20]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f+20] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f+20] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:29: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of

0.4507457106821741

In [195]:
# 10+7,8월+10
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-10]) == '20190911':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-10] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:18: SettingWithCopyWarning: 
A value is trying to be set on 

0.4595823641396911

In [196]:
# 20+7,8월+20
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-20]) == '20190921':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-20] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-20] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:26: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a c

0.4828841772482094

In [190]:
# 6,7,8 월
g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # Remove the CWD from sys.path while we load stuff.
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:28: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  app.launch_new_instance()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:12: S

0.506230318518663


<카테고리별>

1위
2019-6-20 2019-8-31

2위
2019-7-20 2019-8-31

3위
2019-7-10 2019-8-31

<마더코드별>

1위
2019-06-20 2019-08-31

2위
2019-06-10 2019-08-31

3위
2019-06-01 2019-08-31

<마더코드 일치비율>

1위
2019-06-10 2019-09-20

2위
2019-06-20 2019-09-10

3위
2019-05-10 2019-08-20

In [197]:
#10+7,8월 +5

g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-10]) == '20190906':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-10] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:19: SettingWithCopyWarning: 

0.45547805725959367

In [198]:
#10+7,8월 -5


g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-10]) == '20190826':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-10] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-10] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:24: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  app.launch_new_instance()
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:20: SettingWithCopyWarning: 
A value is trying to be set

0.48643572001220725

In [199]:
#5+7,8월 



g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-5]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-5] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-5] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10


C:\Anaconda\lib\site-packages\ipykernel_launcher.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:21: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:31: SettingWithCopyWarning: 
A value is trying

0.4511867040896017

In [200]:
#15+7.8월


g=0
data_2020['compare']=0
for i in range(len(panda_2019)):
    if str(panda_2019['DATE'][i+a+b+c+d+e+f-15]) == '20190901':
        break
    else:
        g+=1
        if panda_2019['상품군'][i+a+b+c+d+e+f-15] == '의류':
            data_2020['compare'][0]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '속옷':
            data_2020['compare'][1]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '이미용':
            data_2020['compare'][2]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '주방':
            data_2020['compare'][3]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '농수축':
            data_2020['compare'][4]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '무형':
            data_2020['compare'][5]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '가전':
            data_2020['compare'][6]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '생활용품':
            data_2020['compare'][7]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '가구':
            data_2020['compare'][8]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '잡화':
            data_2020['compare'][9]+=1
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '건강기능':
            data_2020['compare'][10]+=1 
        elif panda_2019['상품군'][i+a+b+c+d+e+f-15] == '침구':
            data_2020['compare'][11]+=1     
month7_8=data_2020['비율']-data_2020['compare']/g
month7_8_10=0
for i in range(len(month7_8)):
    month7_8_10+=abs(month7_8[i])
month7_8_10

C:\Anaconda\lib\site-packages\ipykernel_launcher.py:28: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:24: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Anaconda\lib\site-packages\ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a

0.4508696161658707